lib/jni/PrivacyGuard_jni.cpp

```diff
@@ -143,14 +120,14 @@ Java_com_microsoft_applications_events_PrivacyGuard_nativeAddIgnoredConcern(JNIE
         jstring eventName,
         jstring fieldName,
         jint dataConcern) {
-    if (spPrivacyGuard == nullptr) {
+    if (!PrivacyGuardState::isPrivacyGuardInstanceInitialized()) {
         return;
     }
 
     auto eventNameStr = JStringToStdString(env, eventName);
     auto fieldNameStr = JStringToStdString(env, fieldName);
     auto dataConcernInt = static_cast<uint8_t>(dataConcern);
-    spPrivacyGuard->AddIgnoredConcern(eventNameStr, fieldNameStr, static_cast<DataConcernType >(dataConcernInt));
+    PrivacyGuardState::getPrivacyGuardInstance()->AddIgnoredConcern(eventNameStr, fieldNameStr, static_cast<DataConcernType >(dataConcernInt));
```

Comment:
What if this returns nullptr?

Reply:
Why not have it return a &?

Reply:
I do want the ref-count to be incremented for the shared_ptr to prevent any fun issues with concurrency. There is guards in place to ensure the value being updated safely, but I want the caller to be able to have a ref to work with as well. (I've refactored a bit to make this more clear.)